src/app/services/constraint.service.ts

```diff
@@ -124,6 +131,8 @@ export class ConstraintService {
   // [pDroppable] in gb-cross-table.component
   public variablesDragDropScope = 'PrimeNGVariablesDragDropContext';
 
+  private _variablesViewMode: VariablesViewMode;
```

Comment:
At first sight, this does not look like the right place to store this view mode. But then, it looks like this service already has gotten out of hand, it has too many responsibilities already.